pkg/gui/controllers/helpers/refs_helper.go

```diff
@@ -133,31 +137,86 @@ func (self *RefsHelper) ResetToRef(ref string, strength string, envVars []string
 	return nil
 }
 
-func (self *RefsHelper) CreateGitResetMenu(ref string) error {
+func (self *RefsHelper) CreateGitResetMenu(ref string, refType string) error {
 	type strengthWithKey struct {
 		strength string
 		key      types.Key
+		ref      string
 	}
+
 	strengths := []strengthWithKey{
-		{strength: "soft", key: 's'},
-		{strength: "mixed", key: 'm'},
-		{strength: "hard", key: 'h'},
+		{strength: "soft", key: 's', ref: ref},
+		{strength: "mixed", key: 'm', ref: ref},
+		{strength: "hard", key: 'h', ref: ref},
+	}
+
+	// File controller already passes "@{upstream}" as the ref by default,
+	// so this is to avoid duplication
+	if refType != "file" {
+		strengths = append(strengths, []strengthWithKey{
+			{strength: "soft", key: 'S', ref: "@{upstream}"},
+			{strength: "mixed", key: 'M', ref: "@{upstream}"},
+			{strength: "hard", key: 'H', ref: "@{upstream}"},
```

Comment:
This clashes with the global horizontal scrolling mapping :)

Reply:
damn that's annoying. Maybe 'u'?

Reply:
I'd rather go in a different direction, this specific mapping has bitten me in the butt as well.

I don't have a proposal as to which way should we go, but I know backwards compatibility is (supposed to be) _a thing._